learn2learn/vision/models.py

```diff
@@ -168,6 +168,28 @@ def forward(self, x):
         return super(OmniglotFC, self).forward(x.view(-1, self.input_size))
 
 
+class OmniglotFCOML(nn.Sequential):
+    """
+    Adaptation for OML
+    """
+
+    def __init__(self, input_size, output_size, sizes=None):
+        if sizes is None:
+            sizes = [256, 128, 64, 64]
+        layers = [LinearBlock(input_size, sizes[0]), ]
+        for s_i, s_o in zip(sizes[:-1], sizes[1:]):
+            layers.append(LinearBlock(s_i, s_o))
+        # break down into features and adaptaion
+        self.features = nn.Sequential(*layers)
+        self.adaptation = fc_init_(nn.Linear(sizes[-1], output_size))
```

Comment:
This network separates adaptation params from the feature extractor, making it easier to freeze later.

Reply:
A cleaner way to make this extensible to more models is to imbue the model with a `get_adaptation_params` function, and on the experiment side we check if the given model has this function in the setup before proceeding.

That should also make it easier to test different versions of the model where more or less layers are frozen during the inner loop, as it is done in the OML and ANML papers.